src/main/java/cloud/orbit/spring/actuate/ActorInfoContributorLifetimeExtension.java

```diff
@@ -0,0 +1,176 @@
+
+    ActorInfoContributorLifetimeExtension(final OrbitRemoteReferenceService orbitRemoteReferenceService,
+                                          final LifetimeExtension springLifetimeExtension)
+    {
+        this.orbitRemoteReferenceService = orbitRemoteReferenceService;
+        this.springLifetimeExtension = springLifetimeExtension;
+    }
+
+    @Override
+    public <T> T newInstance(final Class<T> concreteClass)
+    {
+        // In case Orbit chooses this extension as the "primary" LifetimeExtension, we need to redirect to the
+        // SpringLifetimeExtension for constructing new instances
+        return springLifetimeExtension.newInstance(concreteClass);
+    }
+
+    @Override
+    public Task<?> preActivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::add);
+    }
+
+    @Override
+    public Task<?> preDeactivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::remove);
+    }
+
+    private Task<?> doIfInfoContributor(final AbstractActor<?> actor,
+                                        final Consumer<ActorInfoContributorReference> synchronizedAction)
+    {
+        if (actor instanceof InfoContributor)
+        {
+            ActorInfoContributorReference actorInfoContributorReference = new ActorInfoContributorReference(actor);
+            synchronized (actorInfoContributorReferences)
+            {
+                synchronizedAction.accept(actorInfoContributorReference);
+            }
+        }
+        return Task.done();
+    }
+
+    @Override
+    public void contribute(final Info.Builder builder)
+    {
+        createActorInfoObject().ifPresent(actorInfoObject -> builder.withDetail("actors", actorInfoObject));
+    }
+
+    private Optional<?> createActorInfoObject()
+    {
+        Set<ActorInfoContributorReference> actorInfoContributorReferencesCopy;
+        synchronized (actorInfoContributorReferences)
+        {
+            actorInfoContributorReferencesCopy = actorInfoContributorReferences.stream()
+                    .map(ActorInfoContributorReference::new)
+                    .collect(Collectors.toSet());
+        }
+        Collection<ActorInfoContributorReference> expiredReferences = new ArrayList<>();
+        Map<String, Map<String, Object>> actorInfoObject = new HashMap<>();
+        actorInfoContributorReferencesCopy.forEach(actor ->
+        {
+            InfoContributor contributor = actor.reference.get();
+            if (contributor == null)
+            {
+                expiredReferences.add(actor);
+            }
+            else
+            {
+                final Info.Builder builder = new Info.Builder();
+                contributor.contribute(builder);
+                actorInfoObject.putIfAbsent(actor.name, new HashMap<>());
+                actorInfoObject.get(actor.name).put(actor.identity, builder.build().getDetails());
```

Comment:
Well this is kind of a tough design call. Actors are in some ways fundamentally different than a typical service or component class since they conceptually always exist and can be spun up with different identities. I made the choice to dictate the structure of root->actorInterface->actorIdentity for a couple reasons.

First, I wanted to mitigate the chances of bugs turning up where, for example, a user has two actor instances that both contribute some state data to the builder. Whichever actor was visited last would "win" and mask the first actor, since they have different identities.

Second, and perhaps less valid, is I can't think of a use case where a user _wouldn't_ want this kind of breakdown of their info, so I was trying to add built-in functionality that users would otherwise have to build themselves, i.e.:
```java
@Override
public void contribute(final Info.Builder builder) {
    builder.withDetail(String.format("%s/%s/data", "MyActor", getIdentity()), state().getData());
}
```
which is inelegant and completely loses the json structure due to the limitations of the `Info.Builder` interface.

Now, we could add a config property to let the user choose which behavior they want, like so:

```yaml
management.info.actors.structure: NORMAL | BY_IDENTITY
```

Thoughts?